src/prebid.js

```diff
@@ -100,6 +100,20 @@ function validateSizes(sizes, targLength) {
   return cleanSizes;
 }
 
+function setBattrForAdUnit(adUnit, mediaType) {
+  const ortb2Imp = adUnit.ortb2Imp || {};
+  const mediaTypes = adUnit.mediaTypes || {};
+
+  if (ortb2Imp[mediaType]?.battr && mediaTypes[mediaType]?.battr && (ortb2Imp[mediaType]?.battr !== mediaTypes[mediaType]?.battr)) {
+    logWarn('battr field differ between ortb2Imp and mediaTypes');
+  }
+
+  const battr = ortb2Imp[mediaType]?.battr || mediaTypes[mediaType]?.battr;
+  
+  adUnit.ortb2Imp = {...ortb2Imp, [mediaType]: {...(ortb2Imp[mediaType] || {}), battr}};
+  adUnit.mediaTypes = {...mediaTypes, [mediaType]: {...(mediaTypes[mediaType] || {}), battr}};
```

Comment:
I think this would add `battr: undefined` in most cases. You can also simplify a bit by using `deepSetValue`, e.g.

```suggestion
  if (battr != null) {
      ['mediaTypes', 'ortb2Imp'].forEach(target => deepSetValue(adUnit, `${target}.${mediaType}.battr`, battr))
  }
```

could you please also add some tests?